meta-iotconnect/recipes-apps/temp/files/0001_CMake_findPackage.patch

```diff
@@ -0,0 +1,24 @@
+diff --git a/iotc-generic-c-sdk/CMakeLists.txt b/iotc-generic-c-sdk/CMakeLists.txt
+index a3d45c0..17d38c3 100644
+--- a/iotc-generic-c-sdk/CMakeLists.txt
++++ b/iotc-generic-c-sdk/CMakeLists.txt
+@@ -84,15 +84,7 @@ ELSE ()
+     )
+ ENDIF ()
+ 
+-
+-IF (CMAKE_TOOLCHAIN_FILE)
```

Comment:
Why are we even patching this at this level? We can just modify the SDK to switch on whatever condition with a CMake define, in order to do what you need. If we change the genefic C SDK makefile on that end, this patch may not work anymore.

Reply:
There a as-yet-unresolved issue that the "find_package" mechanism in a CMake project built within Yocto can resolve to host-native libraries instead of the target libraries.

This patch is a temporary fix to remove that hurdle for now.

Reply:
And what I am saying is that you implement this unresolved issue workaround in the main project - pretty much the same way you are patching it. I don't see how is what I am suggesting not ovious:

Pseudocode (not sure exatly how this would go):
```cmake
IF (PC_CURL_LIBRARY_DIRS)
    target_link_libraries(iotc-c-generic-sdk ${PC_CURL_LIBRARY_DIRS}/libcurl.so)
ELSEIF (CMAKE_TOOLCHAIN_FILE)
    # not the best way to detect VCPKG, but we'll go with that
    find_package(CURL CONFIG REQUIRED)
    target_link_libraries(iotc-c-generic-sdk CURL::libcur)
ELSE ()
    find_package(CURL REQUIRED)
    target_link_libraries(iotc-c-generic-sdk CURL::libcur)
ENDIF()

target_link_libraries(iotc-c-generic-sdk cjson) 
``` 

That  way any (generic-c-sdk) customer that has issues linking curl can just do the same.